### DISCUSSION TITLE
Error validating datasource `db`: the URL must start with the protocol `mongo`.

### DISCUSSION THREAD

**Discussion by Chinzillla** [prisma/prisma#16734] [Q&A] [answered]
### Bug description

```
`Invalid prisma.chat.findMany() invocation:

error: Error validating datasource `db`: the URL must start with the protocol `mongo`.
  -->  schema.prisma:10
     | 
  9 |   provider = "mongodb"
10 |   url      = env("DATABASE_URL")
     | 

Validation Error Count: 1
    at RequestHandler.handleRequestError (/var/task/node_modules/@prisma/client/runtime/index.js:34883:13)
    at RequestHandler.handleAndLogRequestError (/var/task/node_modules/@prisma/client/runtime/index.js:34841:12)
    at /var/task/node_modules/@prisma/client/runtime/index.js:35387:25
    at async PrismaClient._executeRequest (/var/task/node_modules/@prisma/client/runtime/index.js:35955:22)
    at async PrismaClient._request (/var/task/node_modules/@prisma/client/runtime/index.js:35926:16)
    at async getServerSideProps (/var/task/.next/server/pages/GuestBook.js:419:18)
    at async Object.renderToHTML (/var/task/node_modules/next/dist/server/render.js:506:20)
    at async doRender (/var/task/node_modules/next/dist/server/base-server.js:708:34)
    at async cacheEntry.responseCache.get.isManualRevalidate.isManualRevalidate (/var/task/node_modules/next/dist/server/base-server.js:813:28)`
```

**What I have tried:**
Different type of methods I've tried so far:

1. Renaming mongodb to mongo in database URL
2. Removing the quotes in vercel enviroment variables
3. adding the enviroment variable to vercel.json
4. Works perfectly fine in Dev -> Build -> Start in vscode.

**My Question:**
How do you deploy on vercel with prisma mongodb URL error? Ive seen the same error for others but with different db.

### How to reproduce

1. npm run dev
2. npm run build
3. npm run start - Connects to Mongodb successfully
4. Deploy on Vercel
5. Successfully builds
6. Route times out and prints out error above.

### Prisma Schema File:
----------------------------------------------------------------------------------------

```
// This is your Prisma schema file,
// learn more about it in the docs: https://pris.ly/d/prisma-schema

generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider = "mongodb"
  url      = env("DATABASE_URL")
}
model Viewer {
  id String @id @default(auto()) @map("_id") @db.ObjectId
  counter Int @default(0)
}

model Chat {
  id            String    @id @default(auto()) @map("_id") @db.ObjectId
  name          String
  text          String
  createdAt  DateTime   @default(now())
}
```

### Next.js Page:
----------------------------------------------------------------------------------------

```

import AuthNav from "../components/Auth"
import Link from "next/link"
import Button from "../components/Button"
import { useSession } from "next-auth/react"
import DisplayForm from "../components/GuestBookForm"
import prisma from '../../lib/prisma'
import { makeSerializable } from '../../lib/util'
import DisplayList from "../components/GuestBookListDisplay"


export default function GuestBookPage({ chat }) {
  const { data } = useSession()
  const random = ""

  if (data) {
    return (
      <div>
        <AuthNav></AuthNav>
        <Button className="mt-10 border-2 dark:border-white border-black hover:bg-gray-900 hover:text-white dark:hover:bg-white dark:hover:text-black">
          <Link href="/api-example">Check your data</Link>
        </Button>
        <ul className="flex flex-col mt-10">
          <DisplayForm />
          <DisplayList Data={chat}/>
        </ul>
      </div>
    )
  } else {
    return (
      <div className="flex">
        <AuthNav></AuthNav>
        <ul className="flex flex-col mt-10">
        <DisplayList Data={chat}/>
        </ul>
      </div>
    )
  }
}


export const getServerSideProps = async () => {
  const chat = await prisma.chat.findMany({
    orderBy: {
      createdAt: 'desc',
    }
  })
  return {
    props: { chat:  makeSerializable(chat) },
  }
}
```

### Prisma Version

```
"prisma": "^4.7.1",
```

**Comment by janpio** [comment]
What is output when you try to log `process.env.DATABASE_URL`?

**Reply by Chinzillla** [reply]
mongodb+srv://xxx:xxx@xxx.xxx.mongodb.net/%3FretryWrites=true&w=majority

**Reply by Chinzillla** [reply]
That is whats stored in Vercel env

**Reply by janpio** [reply]
This technically misses a database name being the `mongodb.net`. This error message would be surpsing though anyway - but maybe see what happens when you just add that.

**Comment by Chinzillla** [comment, marked as answer]
The solution I found was that a duplicate Prisma Schema was generated in the node modules for some reason. When I deleted it and redeployed it on vercel it worked. I also removed the " " from the database URL within vercel environment variables.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
